cmd/runtime_options.go

```diff
@@ -52,13 +53,21 @@ func runtimeOptionFlagSet(includeSysEnv bool) *pflag.FlagSet {
 	flags := pflag.NewFlagSet("", 0)
 	flags.SortFlags = false
 	flags.Bool("include-system-env-vars", includeSysEnv, "pass the real system environment variables to the runtime")
+	// NOTE(imiric): This flag should only be defined in cmd.optionFlagSet,
+	// yet because of the fragmented JS runtime options configuration
+	// (and flag handling?) the cmd.TestEnvVars test would fail if this
+	// wasn't redefined here. Functionally it shouldn't matter, but this
+	// should be fixed as part of #883.
```

Comment:
Can't you just remove it from the other flags ?

Reply:
If I remove it from `optionFlagSet()`, then `TestConfigCmd` fails. :disappointed: 

I hesitate to edit the test to do `fs.AddFlagSet(runtimeOptionFlagSet(false))`, but let me know if you think that would be an improvement.

Reply:
I'm confused - `CompatibilityMode` should only be in `lib.RuntimeOptions`, it doesn't make sense to be both there and in `lib.Options`. Why would `TestConfigCmd` or `TestEnvVars` fail if it's not in `lib.Options`?

Reply:
Hhmm well, you're right, it was redundant in `lib.Options`, as I removed it in 2b3c9c31, and everything still works, so yay... :)

But I definitely noticed some weird test behavior before where they had to be specified in both places or one of two tests would fail. Anyway, it should be fixed now.